### PR TITLE
Milestone 6: Friendships v2

### PR DESCRIPTION
**Pull request by diegoarvz4**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -16,4 +16,23 @@ def configure_permitted_parameters
                profile_attributes: %i[id name city country description])
     end
   end
+
+  def find_friends
```
Comment (efrapp) [inline app/controllers/application_controller.rb]: This method isn't the responsibility of the `ApplicationController`, it should be handled by the User or Friendship classes. I understand you need it in the PostController too but you need to find a different way to use it without sacrificing the Single Responsibility Principle.

File: app/models/friendship.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal:true
+
+class Friendship < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+  validates :user_id, uniqueness: { scope: :friend_id }
```
Comment (mcrd25) [inline app/models/friendship.rb]: I don't think this does what you think it does. I am able to create two friend requests essentially. When I have already sent one the other user, the receiver can receive. This is a basic validation but based on your implementation of friendships you need more.

Se below:
Screen Shot 2019-09-16 at 6 00 25 PM
Screen Shot 2019-09-16 at 6 00 39 PM

What this does is prevent something like:
Screen Shot 2019-09-16 at 6 10 30 PM

This is fine. But, based on your implementation of friendship more would need to be done. Also, at basic level, you can't allow a person to send a friend request to someone that's already sent them a friend request.

Comment (mcrd25) [reply app/models/friendship.rb]: [OPTIONAL] [STRONGLY RECOMMENDED]
What I will say is that maybe you can try using the hint provided for a milestone in handling mutual friendships. Creating an additional column upon confirming a friendship may be necessary. [OPTIONAL]
 See links mentioned in hint for milestone:
https://explainextended.com/2009/03/07/selecting-friends/
https://stackoverflow.com/questions/4219979/sql-best-practice-for-a-friendship-table

As mentioned in pathwright
Screen Shot 2019-09-16 at 4 57 03 PM

**Comment by diegoarvz4** [comment]
# Summary of changes made
- Users friend methods move to a friends.rb module in the bin folder.
- Mutual friendships validations and tests
- Posts bug fixed when login/sign in

**Pull request merged**